src/zombiereborn.cpp

```diff
@@ -178,34 +232,132 @@ void ZR_FakePlayerDeath(CCSPlayerController *pAttackerController, CCSPlayerContr
+
+	pVictimPawn->SetModel("characters/models/tm_phoenix/tm_phoenix.vmdl");
 }
 
 
 void ZR_InfectMotherZombie(CCSPlayerController *pVictimController)
 {
-	ZR_FakePlayerDeath(pVictimController, pVictimController, "prop_exploding_barrel"); // or any other killicon
 	pVictimController->SwitchTeam(CS_TEAM_T);
 
-	//set model/health/etc here
+	CCSPlayerPawn *pVictimPawn = (CCSPlayerPawn*)pVictimController->GetPawn();
+	if (!pVictimPawn)
+		return;
+	// set model/health/etc here
+	
+	//hardcode everything for now
+	pVictimPawn->m_iMaxHealth = 10000;
+	pVictimPawn->m_iHealth = 10000;
+
+	pVictimPawn->SetModel("characters/models/tm_phoenix/tm_phoenix.vmdl");
 }
 
+void ZR_InitialInfection()
+{
+	// grab player count and mz infection candidates
+	CUtlVector<CCSPlayerController*> pCandidateControllers;
+	int iPlayerCount = 0;
+	for (int i = 0; i < gpGlobals->maxClients; i++)
+	{
+		CCSPlayerController* pController = CCSPlayerController::FromSlot(i);
+		if (!pController)
+			continue;
+		iPlayerCount++;
+		if (pController->m_iTeamNum() != CS_TEAM_CT)
+			continue;
+
+		CCSPlayerController* pPawn = (CCSPlayerController*)pController->GetPawn();
+		if (!pPawn || !pPawn->IsAlive())
+			continue;
+
+		pCandidateControllers.AddToTail(pController);
+	}
+
+	// calculate the num of mz to infect
+	int iMZToInfect = iPlayerCount / g_flInfectSpawnMZRatio;
+	iMZToInfect = g_flInfectSpawnMinCount > iMZToInfect ? g_flInfectSpawnMinCount : iMZToInfect;
+
+	// get spawn points
+	CUtlVector<SpawnPoint*> spawns;
+	if (g_flInfectSpawnType == EZRSpawnType::RESPAWN)
+	{
+		SpawnPoint* spawn = nullptr;
+		while (nullptr != (spawn = (SpawnPoint*)UTIL_FindEntityByClassname(spawn, "info_player_*")))
+		{
+			if (spawn->m_bEnabled())
+				spawns.AddToTail(spawn);
+		}
+	}
+
+	// infect
+	while (pCandidateControllers.Count() > 0 && iMZToInfect > 0)
+	{
+		int randomindex = rand() % pCandidateControllers.Count();
+		ZR_InfectMotherZombie(pCandidateControllers[randomindex]);
+
+		CCSPlayerController* pPawn = (CCSPlayerController*)pCandidateControllers[randomindex]->GetPawn();
+		if (pPawn && spawns.Count())
+		{
+			int randomindex = rand() % spawns.Count();
+			pPawn->SetAbsOrigin(spawns[randomindex]->GetAbsOrigin());
+			pPawn->SetAbsRotation(spawns[randomindex]->GetAbsRotation());
+		}
+
+		pCandidateControllers.FastRemove(randomindex);
+		iMZToInfect--;
+	}
+	ClientPrintAll(HUD_PRINTCENTER, "First infection has started!");
+	ClientPrintAll(HUD_PRINTTALK, ZR_PREFIX "First infection has started! Good luck, survivors!");
+	g_ZRRoundState = EZRRoundState::POST_INFECTION;
+}
+
+void ZR_StartInitialCountdown()
+{
+	int iRoundNum = g_iRoundNum;
+	g_iInfectionCountDown = g_flInfectSpawnTimeMin + (rand() % (g_flInfectSpawnTimeMax - g_flInfectSpawnTimeMin + 1));
+	new CTimer(1.0f, false, [iRoundNum]()
+	{
+		if (iRoundNum != g_iRoundNum)
+			return -1.0f;
+		if (g_iInfectionCountDown <= 0)
+		{
+			ZR_InitialInfection();
+			return -1.0f;
+		}
+
+		if (g_iInfectionCountDown <= 15)
```

Comment:
I know we used this value in Lua ZR, but we should consider a higher value that won't make this check fail with reasonable infection time configs. 1 minute maybe?